composables/useTransaction.ts

```diff
@@ -151,32 +149,35 @@ const useExecuteTransaction = (options: TransactionOptions) => {
       txHash.value = param.txHash || undefined
     }
 
-    execByVm({
-      SUB: () => {
-        ;(howAboutToExecute as SubstrateHowAboutToExecute)(
-          accountId.value,
-          (params as ExecuteSubstrateTransactionParams).cb,
-          arg,
-          {
+    doAfterCheckCurrentChainVM(() => {
```

Comment:
I get this approach but I'm not to sold on adding this inside `useTransaction`, I'd rather show the modal before the action has started.

maybe because this is simplest solution atm? 

<details><summary>see</summary>

just a visual examaple of where this would run after action has started ,

![CleanShot 2024-08-15 at 10 54 00@2x](https://github.com/user-attachments/assets/9b638289-1423-4f2c-8f07-793398cb0794)

</details>



 @preschian wdyt

Reply:
> I'd rather show the modal before the action has started.
> 
> maybe because this is simplest solution atm?

Yes, it's the simplest solution, then we don't need to add this check everywhere.

although I have added checks for most cases such as mint/drop/buy, I leave it here to make sure it is impossible to do on-chain operations with an incorrect wallet.